File: app/src/main/java/com/example/paetz/yacguide/utils/DateUtils.java
package com.example.paetz.yacguide.utils;

import java.text.ParseException;
import java.text.SimpleDateFormat;
import java.util.Calendar;
import java.util.Date;

public class DateUtils {

    public final static String UNKNOWN_DATE = "0000-00-00";

    public static String formatDate(String date) {
        SimpleDateFormat format = new SimpleDateFormat("yyyy-MM-dd");
        try {
            Date formattedDate = format.parse(date);
            Calendar cal = Calendar.getInstance();
            cal.setTime(formattedDate);
            return cal.get(Calendar.DAY_OF_MONTH) + "." + (cal.get(Calendar.MONTH) + 1) + "." + cal.get(Calendar.YEAR);
        } catch (ParseException e) {
            return "";
        }
    }
}

File: app/src/main/java/com/yacgroup/yacguide/utils/DateUtils.kt
/*
 * Copyright (C) 2019 Fabian Kantereit
 *
 * This program is free software: you can redistribute it and/or modify
 * it under the terms of the GNU General Public License as published by
 * the Free Software Foundation, either version 3 of the License, or
 * (at your option) any later version.
 *
 * This program is distributed in the hope that it will be useful,
 * but WITHOUT ANY WARRANTY; without even the implied warranty of
 * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
 * GNU General Public License for more details.
 *
 * You should have received a copy of the GNU General Public License
 * along with this program.  If not, see <https://www.gnu.org/licenses/>.
 */

package com.yacgroup.yacguide.utils

import android.annotation.SuppressLint
import java.text.ParseException
import java.text.SimpleDateFormat
import java.util.Calendar

object DateUtils {

    const val UNKNOWN_DATE = "0000-00-00"

    @SuppressLint("SimpleDateFormat")
    fun formatDate(date: String): String {
        val format = SimpleDateFormat("yyyy-MM-dd")
        return try {
            val formattedDate = format.parse(date)
            val cal = Calendar.getInstance()
            cal.time = formattedDate
            "${cal.get(Calendar.DAY_OF_MONTH)}.${cal.get(Calendar.MONTH) + 1}.${cal.get(Calendar.YEAR)}"
        } catch (e: ParseException) {
            ""
        }

    }
}
